roles/kafka_broker/tasks/kraft_tasks.yml

```
@@ -0,0 +1,17 @@
+---
+- name: extract uuid from meta.properties
+  slurp:
+    src: "{{ kafka_controller_final_properties['log.dirs'] }}/meta.properties"
+  delegate_to: "{{ groups.kafka_controller[0] }}"
+  register: uuid_broker
+
+- set_fact:
+    uuid_key_broker: "{{ uuid_broker['content'] | b64decode}} "
+
+- name: get CLUSTER_ID
+  set_fact:
+    clusterid: "{{ uuid_key_broker.partition('cluster.id=')[2].partition('\n')[0] }}"
+
+- name: format meta directory
+  shell: "{{ binary_base_path }}/bin/kafka-storage  format -t {{ clusterid }} -c {{ kafka_broker.config_file }} --ignore-formatted"
+  register: format_meta
```

```suggestion
- name: Format Storage Directory
  shell: "{{ binary_base_path }}/bin/kafka-storage  format -t {{ clusterid }} -c {{ kafka_broker.config_file }} --ignore-formatted"
  register: format_meta
  with_fact:
    clusterid: "{{ uuid_key_broker.partition('cluster.id=')[2].partition('\n')[0] }}"
```

You should be able to remove the get CLUSTER_ID task via this ^^

It may even be possible to get rid of both set_facts, id toy around with it
```
  with_fact:
    clusterid: "{{ (uuid_broker['content'] | b64decode).uuid_key_broker.partition('cluster.id=')[2].partition('\n')[0] }}"
```